app/core/abilities/abilities.ts

```diff
@@ -9081,6 +9081,53 @@ export class RockHeadStrategy extends AbilityStrategy {
   }
 }
 
+export class FairyLockStrategy extends AbilityStrategy {
+  process(
+    pokemon: PokemonEntity,
+    state: PokemonState,
+    board: Board,
+    target: PokemonEntity,
+    crit: boolean
+  ) {
+    super.process(pokemon, state, board, target, crit)
+    
+    const cells = board.getAdjacentCells(
+      target.positionX,
+      target.positionY,
+      true
+    )
+
+    let inhabitedCells = new Array<Cell>()
+    let enemiesHit = 0
+
+    cells.forEach((cell) => {
+      if (cell.value && cell.value.team !== pokemon.team){
+        inhabitedCells.push(cell)
+        enemiesHit += 1
+      }
+    })
+    inhabitedCells.forEach((cell) => {
+      if (cell.value && cell.value.team !== pokemon.team) {
+        pokemon.simulation.room.broadcast(Transfer.ABILITY, {
+          id: pokemon.simulation.id,
+          skill: pokemon.skill,
+          targetX: cell.value.positionX,
+          targetY: cell.value.positionY
+        })
+        cell.value.handleSpecialDamage(
+          Math.round(90 / enemiesHit),
+          board,
+          AttackType.SPECIAL,
+          pokemon,
+          crit
+        )
+      }
+    })
+    
+    target.status.triggerLocked(1500 * enemiesHit, target)
```

Comment:
i'm not a fan of this design, scaling with nb of enemies hit. Feels like a double penalty and drastically change the strength of the move. i would prefer a fixed delay. We learned the lesson in the past that all the hard CC status effects durations should not scale